Mage/src/main/java/mage/watchers/common/NumberOfTimesPermanentTargetedATurnWatcher.java

```diff
@@ -25,24 +26,23 @@ public NumberOfTimesPermanentTargetedATurnWatcher() {
 
     @Override
     public void watch(GameEvent event, Game game) {
-        if (event.getType() == GameEvent.EventType.TARGETED) {
-            Permanent permanent = game.getPermanent(event.getTargetId());
-            if (permanent != null) {
-                MageObjectReference mor = new MageObjectReference(permanent, game);
-                int amount = 0;
-                if (permanentsTargeted.containsKey(mor)) {
-                    amount = permanentsTargeted.get(mor);
-                }
-                permanentsTargeted.put(mor, ++amount);
-            }
+        if (event.getType() != GameEvent.EventType.TARGETED) {
+            return;
+        }
+        StackObject targetingObject = CardUtil.getTargetingStackObject(event, game);
+        if (targetingObject == null || !CardUtil.checkTargetMap(null, targetingObject, event, game)) {
```

Comment:
I think it does need to write even with the watcher (it's necessary to prevent the bug from #8802) counting a select/deselect as first time targeted that turn). As I understand it, Java converts the null object to "null" string before concatenating and stores the key "targetMapnull". You're right that I should add an explanation comment about that usage in the watcher. I could also try to initialize the watcher with a random UUID and use that, might be better?

Reply:
I dont like that watcher works like a trigger and modify targetMap values outside of the watcher. Its will be broken if you add another watcher with same targetMap usage. If you use same code style then targetMap must contains some prefix instead null to split it between triggers/watchers/etc.

Example:
* checkTargetMap(“watcher_name_1”, …
* checkTargetMap(“watcher_name_2”, …
* checkTargetMap(“triggers”, …

Reply:
Hmm, also prefix can help with usage docs like:
checkTargetMap(“trigger_once”

Reply:
@xenohedron found interesting comment about `checkTrigger` method. Looks like it can broke `targetMap`:

![shot_230926_104814](https://github.com/magefree/mage/assets/8344157/abe14df8-758d-4562-8609-cf92b21de5b7)

Reply:
Interesting... I don't think it's currently broken, but I am researching

Reply:
See #11212 for my proposed solution